src/vs/workbench/contrib/preferences/browser/preferencesActions.ts

```diff
@@ -77,3 +79,21 @@ CommandsRegistry.registerCommand({
 	}
 });
 
+//#region --- Register a command to get all actions from the command palette
+CommandsRegistry.registerCommand('_getAllCommands', function () {
+	const actions: { command: string; label: string; precondition?: string }[] = [];
+	for (const editorAction of EditorExtensionsRegistry.getEditorActions()) {
+		actions.push({ command: editorAction.id, label: editorAction.label, precondition: editorAction.precondition?.serialize() });
+	}
+	for (const menuItem of MenuRegistry.getMenuItems(MenuId.CommandPalette)) {
+		if (isIMenuItem(menuItem)) {
+			const title = typeof menuItem.command.title === 'string' ? menuItem.command.title : menuItem.command.title.value;
+			const category = menuItem.command.category ? typeof menuItem.command.category === 'string' ? menuItem.command.category : menuItem.command.category.value : undefined;
+			const label = category ? `${category}: ${title}` : title;
+			actions.push({ command: menuItem.command.id, label, precondition: menuItem.when?.serialize() });
```

Comment:
one is using precondition, one is using when, but you use the same serialized name